save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/TestSuitesSourcesDisplayer.kt

```diff
@@ -0,0 +1,172 @@
+ */
+fun ChildrenBuilder.showTestSuitesSources(
+    testSuitesSources: TestSuitesSourceDtoWithIdList,
+    selectHandler: (TestSuitesSourceDtoWithId) -> Unit,
+    fetchHandler: (TestSuitesSourceDto) -> Unit,
+    editHandler: (TestSuitesSourceDtoWithId) -> Unit,
+) {
+    div {
+        className = ClassName("list-group")
+        testSuitesSources.forEach {
+            showTestSuitesSourceAsListElement(it, false, selectHandler, editHandler, fetchHandler)
+        }
+    }
+}
+
+/**
+ * Display list of [TestSuitesSourceSnapshotKey] of [selectedTestSuiteSource]
+ *
+ * @param selectedTestSuiteSource
+ * @param testSuitesSourcesSnapshotKeys
+ * @param selectHandler callback invoked on TestSuitesSource selection
+ * @param editHandler callback invoked on edit TestSuitesSource button pressed
+ * @param fetchHandler callback invoked on fetch button pressed
+ * @param deleteHandler callback invoked on [TestSuitesSourceSnapshotKey] deletion
+ */
+@Suppress("LongParameterList", "TOO_MANY_PARAMETERS")
+fun ChildrenBuilder.showTestSuitesSourceSnapshotKeys(
+    selectedTestSuiteSource: TestSuitesSourceDtoWithId,
+    testSuitesSourcesSnapshotKeys: TestSuitesSourceSnapshotKeyList,
+    selectHandler: (TestSuitesSourceDtoWithId) -> Unit,
+    editHandler: (TestSuitesSourceDtoWithId) -> Unit,
+    fetchHandler: (TestSuitesSourceDto) -> Unit,
+    deleteHandler: (TestSuitesSourceSnapshotKey) -> Unit,
+) {
+    ul {
+        className = ClassName("list-group")
+        showTestSuitesSourceAsListElement(selectedTestSuiteSource, true, selectHandler, editHandler, fetchHandler)
+        if (testSuitesSourcesSnapshotKeys.isEmpty()) {
+            li {
+                className = ClassName("list-group-item list-group-item-light")
+                +"This source is not fetched yet..."
+            }
+        } else {
+            li {
+                className = ClassName("list-group-item list-group-item-light")
+                div {
+                    className = ClassName("clearfix")
+                    div {
+                        className = ClassName("float-left")
+                        +"Version"
+                    }
+                    div {
+                        className = ClassName("float-right")
+                        +"Upload time"
+                    }
+                }
+            }
+            testSuitesSourcesSnapshotKeys.forEach { testSuitesSourceSnapshotKey ->
+                li {
+                    className = ClassName("list-group-item")
+                    div {
+                        className = ClassName("clearfix")
+                        div {
+                            className = ClassName("float-left")
+                            +testSuitesSourceSnapshotKey.version
+                        }
+                        buttonBuilder(faTimesCircle, style = null, classes = "float-right btn-sm") {
+                            deleteHandler(testSuitesSourceSnapshotKey)
```

Comment:
do we have a confirmation before deleting?

Reply:
For now we don't, but I can use a simple `window.confirm()` for that, guess it is enough.

Reply:
Done